src/models/right_view/vae/vae_unet_model.py

```diff
@@ -0,0 +1,152 @@
+import torch
+import torch.nn.functional as F
+import pytorch_lightning as pl
+
+from argparse import ArgumentParser
+
+from data.right_data import RightDaVinciDataModule
+from pytorch_lightning.metrics.functional import ssim, psnr
+
+from models.right_view.vae.vae_unet import VariationalUNet
+
+
+class VAEModel(pl.LightningModule):
+    def __init__(
+            self,
+            frames_per_sample: int = 1,
+            frames_to_drop: int = 0,
+            enc_out_dim: int = 128,
+            latent_dim: int = 128,
+            kl_coeff: float = 0.1,
+            lr: float = 0.001,
+            log_tb_imgs: bool = False,
+            tb_img_freq: int = 10000,
+            save_img_freq: int = 50,
+            num_layers: int = 5,
+            features_start: int = 64,
+            bilinear: bool = False,
+            **kwargs
+    ):
+        super().__init__()
+
+        self.frames_per_sample = frames_per_sample
+        self.frames_to_drop = frames_to_drop
+
+        self.save_hyperparameters()
+
+        # TODO: generalize for multi frame
+        self.in_channels = 3
+
+        self.net = VariationalUNet(input_channels=self.in_channels,
+                                   output_channels=3,
+                                   enc_out_dim=enc_out_dim,
+                                   latent_dim=latent_dim,
+                                   num_layers=num_layers,
+                                   features_start=features_start,
+                                   bilinear=bilinear)
+
+    def forward(self, x):
+        return self.net(x)
+
+    def step(self, batch, batch_idx):
+        img, target = batch
+        pred, kl, std_min, std_max = self(img)
+
+        mse_loss = ((pred - target) ** 2).mean(dim=(1, 2, 3))
```

Comment:
why were changes necessary here?

Reply:
basically we want to sum mse + kl for every element in the batch first and then average. we don't want (average mse for all samples in batch) + (average kl for all samples in the batch)

does that make sense?